common/lib/fs-utils.ts

```diff
@@ -44,8 +43,23 @@ export function calculateDirectorySize( directoryPath: string ): Promise< number
 
 export function isWordPressDirectory( projectPath: string ): boolean {
 	return (
-		existsSync( path.join( projectPath, 'wp-content' ) ) &&
-		existsSync( path.join( projectPath, 'wp-includes' ) ) &&
-		existsSync( path.join( projectPath, 'wp-load.php' ) )
+		fs.existsSync( path.join( projectPath, 'wp-content' ) ) &&
+		fs.existsSync( path.join( projectPath, 'wp-includes' ) ) &&
+		fs.existsSync( path.join( projectPath, 'wp-load.php' ) )
 	);
 }
+
+// Compare paths in a case-insensitive manner. `fs.Stats.dev` signifies the device ID, and
+// `fs.Stats.ino` signifies the inode number that uniquely identifies the file or directory.
+// The benefit of this approach over converting the entire path to lowercase is that it respects
+// the current file system's case sensitivity.
+export function arePathsEqual( path1: string, path2: string ) {
+	try {
+		const stats1 = fs.statSync( path.resolve( path1 ) );
```

Comment:
I noticed that we are friquently use `sync` options of functions. In this case I think it's not a problem, but I am thinking - should we have some rule to try to always use async versions, to avoid any potential cases to block event loop?

Reply:
`fs.exists` is deprecated, but `fs.existsSync` is not, so that's why we'd use the sync version of that particular function. As for `fs.statSync`, we could technically use the async version. Given that there's only a single client to this code (this isn't a web server handling multiple clients in parallel), I think we can be more liberal with using sync functions than we would otherwise.

Still, it would be very simple to change these calls to their async counterparts, so I agree it would have been better if I had done that. This code lived in just the CLI to begin with, where the demand for asynchrony is even lower, and I think I just overlooked that part when moving it to `./common`

Reply:
Yeah, totally agree about not having any issues with CLI.
It was more general discussion - to have an agreement inside our team to try to always use async versions, to have consistency and avoid potential cases when some sync version can be unintentionally used inside main app.

Btw, interesting thing - https://stackoverflow.com/questions/31799274/node-js-fs-exists-will-be-deprecated-what-to-use-instead

```
The [io.js docs](https://iojs.org/api/fs.html#fs_fs_exists_path_callback#fs_fs_exists_path_callback) mention the use of fs.stat() or fs.access() in place of fs.exists().
```

Reply:
> some sync version can be unintentionally used inside main app

Since the node.js APIs aren't available in the renderer process, I don't think it's technically possible to block the renderer by using those functions. 

I think we can be pragmatic about using sync functions, but I agree that we should always default to the async versions.